### DISCUSSION TITLE
allow hex code ampersand escapes in HTML and markdown posts

### DISCUSSION THREAD

**Discussion by aescling** [cat-KIBI-glitch/mastodon#17] [Ideas]
so we can use `&#x2019;` instead of `&#8217;`

Edit:

posting the above text (as rendered, not as from the markdown source for this conversation) as a markdown post renders

> so we can use \&#x2019; instead of ’

while parsing `#x2019` as a hashtag. this is silly. we should do as markdown does, namely, rendering that as

> so we can use &#x2019; instead of &#8217;

**Comment by marrus-sh** [comment]
My guess is that this is just a problem with the processing order of things; hashtags are parsed before entities. We should check the CommonMark spec to see if it has guidance on which entities should be supported; if it doesn’t, I think just supporting XML ones are fine. (Note some named HTML entities are not welformed XML entities though.)

This is assuming that Mastodon supports named entities at all; if it only supports CharRefs, that makes our job even easier. Regardless, I would consider this a bug.

(Actually, every aspect of Mastodon’s HTML/Markdown support needs to be reviewed, IMO; it’s largely a giant hack. This is just one in probably a long list of problems.)

**Reply by aescling** [reply]
lol

**Comment by marrus-sh** [comment]
Relevant: glitch-soc/mastodon#1223
Also relevant: glitch-soc/mastodon#1464

**Comment by marrus-sh** [comment]
Issue created for this at #25; continue discussion there!
